flink-clients/src/main/java/org/apache/flink/client/cli/ProgramOptions.java

```diff
@@ -184,10 +175,65 @@ public SavepointRestoreSettings getSavepointRestoreSettings() {
 		return savepointSettings;
 	}
 
-	/**
-	 * Indicates whether the job is a Python job.
-	 */
-	public boolean isPython() {
-		return isPython;
+	public void applyToConfiguration(Configuration configuration) {
+		if (getParallelism() != ExecutionConfig.PARALLELISM_DEFAULT) {
+			configuration.setInteger(CoreOptions.DEFAULT_PARALLELISM, getParallelism());
+		}
+
+		configuration.setBoolean(DeploymentOptions.ATTACHED, !getDetachedMode());
+		configuration.setBoolean(DeploymentOptions.SHUTDOWN_IF_ATTACHED, isShutdownOnAttachedExit());
+		ConfigUtils.encodeCollectionToConfig(configuration, PipelineOptions.CLASSPATHS, getClasspaths(), URL::toString);
+		SavepointRestoreSettings.toConfiguration(getSavepointRestoreSettings(), configuration);
+	}
+
+	public static ProgramOptions create(CommandLine line) throws CliArgsException {
+		if (isPython(line) || containsPythonDependencyOptions(line)) {
+			return createPythonPropramOptions(line);
+		} else {
+			return new ProgramOptions(line);
+		}
+	}
+
+	static boolean isPython(CommandLine line) {
```

Comment:
Should we move these kinds of util methods to a class such as `ProgramOptionsUtils` to make `ProgramOptions` more clean?  What's your thought? @aljoscha